src/db.rs

```diff
@@ -469,6 +478,41 @@ impl Database {
             .await
     }
 
+    /// Runs a database-level command and returns a cursor to the response.
+    pub async fn run_cursor_command(
+        &self,
+        command: Document,
+        options: RunCursorCommandOptions,
+    ) -> Result<Cursor<Document>> {
+        let rcc = RunCommand::new(self.name().to_string(), command, options.selection_criteria.clone(), None)?;
+        let rc_command = RunCursorCommand::new(rcc, options)?; 
+        let client = self.client();
+        client.execute_cursor_operation(rc_command).await
+    }
+
+    /// Runs a database-level command and returns a cursor to the response.
+    pub async fn run_cursor_command_with_session(
+        &self,
+        command: Document,
+        options: impl Into<Option<RunCursorCommandOptions>>,
+        session: &mut ClientSession,
+    ) -> Result<SessionCursor<Document>> {
+        let mut options: Option<RunCursorCommandOptions> = options.into().clone();
+        resolve_selection_criteria_with_session!(self, options, Some(&mut *session))?;
+        let selection_criteria = match options.clone() {
+            Some(options) => options.selection_criteria,
+            None => None,
+        };
+        let option = match options.clone() {
+            Some(options) => options,
+            None => RunCursorCommandOptions::default(),
+        };
```

Comment:
Rather than creating an empty options struct when the user hasn't provided any options, we can just make the `options` field on `RunCursorCommand` an `Option<RunCursorCommandOptions>`. This will save us an allocation when `options` is `None`. (You'll probably need to do similar calls to `and_then` in `handle_response` as I described above to extract fields from the options).